src/main/java/org/jooq/lambda/Seq.java

```diff
@@ -2981,6 +2996,18 @@ public R next() {
     }
 
     /**
+     * Get the index of the first element matching the given predicate.
+     */
+    static <T> OptionalLong indexOf(Iterator<T> iterator, Predicate<? super T> predicate) {
```

Comment:
I would prefer not to make this method publicly available, as I'm not 100% sure about the API design yet. So, I'll move it to `SeqImpl`...